pkgs/by-name/fl/flye/package.nix

```
@@ -0,0 +1,47 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+, zlib
+, curl
+, libdeflate
+, bash
+, coreutils
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "flye";
+  version = "2.9.3";
+
+  src = fetchFromGitHub {
+    owner = "fenderglass";
+    repo = pname;
+    rev = version;
+    hash = "sha256-IALqtIPmvDYoH4w/tk2WB/P/pAcKXxgnsu9PFp+wIes=";
+  };
+
+  nativeCheckInputs = [ python3Packages.pytestCheckHook ];
+  propagatedBuildInputs = [ coreutils ];
+  buildInputs = [
+    zlib
+    curl
+    libdeflate
+  ];
+
+  preConfigure = ''
+    substituteInPlace flye/polishing/alignment.py \
+      --replace "/bin/bash" "${bash}/bin/bash"
```

```suggestion
      --replace-fail "/bin/bash" "${bash}/bin/bash"
```

also maybe consider patchShebang

---

I believe patchShebang wouldn't work in this situation, as it is a call to `/bin/bash` from `subprocess.check_call()` in python and not a shebang.

---

`patchShebang` does same thing as `substituteInPlace`, so I am not sure how its related to `subprocess.check_call()`

---

1. `patchShebang` does not seem to exist.  i assume you mean `patchShebangs`?
2. `patchShebangs` certainly does not do the same thing as `substitueInPlace`.  both modify files in place, but the nature of those modifications are completely different.

`patchShebangs` only patches lines that start with `#!`, whereas `substituteInPlace` only performs the replacements you tell it to.

also, `patchShebangs` is run by default as part of fixupPhase.